frameworks/helloworld/tests/test_nonessential_tasks.py

```diff
@@ -0,0 +1,109 @@
+import logging
+
+import os.path
+import pytest
+
+import sdk_cmd
+import sdk_hosts
+import sdk_install
+import sdk_plan
+import sdk_tasks
+import sdk_utils
+from tests import config
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def configure_package(configure_security):
+    try:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+        options = {
+            "service": {
+                "spec_file": "examples/nonessential_tasks.yml"
+            }
+        }
+
+        sdk_install.install(config.PACKAGE_NAME, config.SERVICE_NAME, 2, additional_options=options)
+
+        yield # let the test session execute
+    finally:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.mark.nick
+@pytest.mark.sanity
+def test_kill_essential():
+    '''kill the essential task, verify that both tasks are relaunched against a matching executor'''
+    verify_shared_executor('hello-0')
+
+    old_ids = sdk_tasks.get_task_ids(config.SERVICE_NAME, 'hello-0')
+    assert len(old_ids) == 2
+
+    sdk_tasks.kill_task_with_pattern(
+        'shared-volume/essential', # hardcoded in cmd, see yml
+        sdk_hosts.system_host(config.SERVICE_NAME, 'hello-0-essential'))
+
+    sdk_tasks.check_tasks_updated(config.SERVICE_NAME, 'hello-0', old_ids) # wait for ids to change...
+    sdk_plan.wait_for_completed_recovery(config.SERVICE_NAME) # ...and for tasks to be up and running
+
+    # the first verify_shared_executor call deleted the files. both should have come back via the relaunch.
+    verify_shared_executor('hello-0', delete_files=False) # leave files as-is for the next test
+
+
+@pytest.mark.nick
```

Comment:
You left some marks around.  Not blocking.  You can use `./test.sh -k test_kill_nonessential` of you want to run just this test, or `./test.sh -k <file_name>` if you want to run just one test file.